.azure-pipelines-steps.yml

```diff
@@ -0,0 +1,27 @@
+steps:
+- script: npm i -g npm@$(npm_version)
+  displayName: Use legacy npm version $(npm_version)
+  condition: ne(variables['npm_version'], '')
+
+- task: NodeTool@0
+  inputs:
+    versionSpec: '$(node_version)'
+  displayName: Use Node $(node_version)
+
+- script: npm install
+  displayName: npm install
+
+- script: npm run coveralls
+  env:
+    COVERALLS_REPO_TOKEN: $(COVERALLS_REPO_TOKEN_SECRET)
```

Comment:
Is this needed?  Currently coveralls just knows that the coverage is being pushed from our public repo.

Reply:
Yes, unfortunately Coveralls isn't as tightly integrated with Pipelines yet. I think that should change going forward, but for now we need to pass in a token to authenticate. You can get that from here:

![image](https://user-images.githubusercontent.com/42773683/54132298-5e005880-43ea-11e9-82ee-26d41f3b191e.png)

I'd also recommend keeping it a [secret variable](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables) to make sure it doesn't get leaked. That's how I have it set up in my configuration as well.

Reply:
Alright.  Yeah, it'd be nice to have deeper coveralls integration.

Do we have all these environment variables that coveralls says we need:
```
CI_NAME
CI_BUILD_NUMBER
CI_BUILD_URL
CI_BRANCH
CI_PULL_REQUEST (optional)
```

Reply:
I'm guessing you're drawing that from [here](https://docs.coveralls.io/supported-ci-services), I believe that's just for Ruby, though I'm not 100% sure - it you look under the "configuration" heading it says "This configuration guide is specifically for the Ruby language. Other language support can be found in the individual READMEs.".

I used the instructions in the [node-coveralls' usage section](https://github.com/nickmerwin/node-coveralls#usage) which say we need COVERALLS_SERVICE_NAME and COVERALLS_REPO_TOKEN. The reason COVERALLS_REPO_TOKEN is specifically called out is that its a secret variable, and secrets aren't scoped into tasks by default (to avoid them leaking to malicious or careless custom tasks). COVERALLS_SERVICE_NAME is also stored in the env though.

![image](https://user-images.githubusercontent.com/42773683/54285399-422abd00-4578-11e9-994d-aad4e60fb4ff.png)

Reply:
As a side note, I'm adding coveralls integration to our backlog, definitely agree this would be a good thing to have.

Reply:
Ah okay!  Thanks for adding it to the backlog 😍